File: web/src/main/java/com/alexeiddg/web/model/AppUser.java
package com.alexeiddg.web.model;

import lombok.*;
import jakarta.persistence.*;
import java.time.OffsetDateTime;

@Entity
@Table(name = "app_user")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AppUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long userId;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "username", nullable = false, unique = true, length = 50)
    private String username;

    @Column(name = "role", nullable = false, length = 20)
    private String role;

    @Column(name = "telegram_id", length = 50)
    private String telegramId;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;
}

File: frontend/src/types/model/AppUser.ts
import { UserRole } from "@/types/enums/UserRole";
import { Team } from "@/types/model/Team";

export interface AppUser {
    id: number;
    name: string;
    username: string;
    telegramId: string;
    email: string;
    userRole: UserRole;
    manager: AppUser;
    team: Team;
    createdAt: string;
    updatedAt: string;
}
